clients/admin-ui/src/features/configure-consent/AddVendor.tsx

```diff
@@ -113,6 +86,55 @@ const AddVendor = ({
       }
     : defaultInitialValues;
 
+  const ValidationSchema = useMemo(
+    () =>
+      Yup.object().shape({
+        name: Yup.string()
+          .required()
+          .label("Vendor name")
+          .test("is-unique", "", (value, context) => {
+            const takenSystemNames = systems
+              .map((s) => s.name)
+              .filter((name) => name !== initialValues.name);
+            if (takenSystemNames.some((name) => name === value)) {
+              return context.createError({
+                message: `You already have a vendor called "${value}". Please specify a unique name for this vendor.`,
+              });
+            }
+            return true;
+          }),
+      }),
+    [systems, initialValues.name]
+  );
+
+  // Subscribe and get dictionary values
+  const { tcf, dictionaryService } = useFeatures();
+  const { isLoading } = useGetAllDictionaryEntriesQuery(undefined, {
+    skip: !dictionaryService,
+  });
+  const dictionaryOptions = useAppSelector(selectAllDictEntries);
+  const lockedForGVL = useAppSelector(selectLockedForGVL);
+
+  const [createSystemMutationTrigger] = useCreateSystemMutation();
+  const [updateSystemMutationTrigger] = useUpdateSystemMutation();
+  const suggestionsState = useAppSelector(selectSuggestions);
+
+  const handleCloseModal = () => {
+    modal.onClose();
+    if (onCloseModal) {
+      onCloseModal();
+    }
+    dispatch(setSuggestions("initial"));
+    dispatch(setLockedForGVL(false));
+  };
+
+  const formRef = useRef(null);
```

Comment:
This is to access the Formik form values from outside the `<Formik>` component, which makes handling the dictionary entries much cleaner (before I was trying to duplicate it in React's state)-- found it [here](https://bionicjulia.com/blog/react-formik-yup-troubleshooting-tips).